code/addons/test/src/node/vitest-manager.ts

```diff
@@ -27,14 +36,31 @@ export class VitestManager {
 
   storyCountForCurrentRun: number = 0;
 
-  constructor(
-    private channel: Channel,
-    private testManager: TestManager
-  ) {}
+  constructor(private testManager: TestManager) {}
 
-  async startVitest(watchMode = false) {
+  async startVitest({ watchMode = false, coverage = false } = {}) {
     const { createVitest } = await import('vitest/node');
 
+    const storybookCoverageReporter: [string, StorybookCoverageReporterOptions] = [
+      '@storybook/experimental-addon-test/internal/coverage-reporter',
+      {
+        testManager: this.testManager,
+        getCoverageOptions: () => this.vitest?.config?.coverage as ResolvedCoverageOptions<'v8'>,
```

Comment:
How come we needed this function for CoverageOptions, but the testManager was fine as is?

Reply:
because the coverage config can change in between runs, if the user changes the config while SB is open, so we can't just pass it in the first time. but now that we're restarting Vitest before each coverage run, that takes care of that problem, so I'll change it.